test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use log::{debug, error};
+use serde::{Deserialize, Deserializer};
+use serde_json::Value;
+use up_rust::{
+    Data, UAttributes, UAuthority, UCode, UEntity, UMessage, UMessageType, UPayload,
+    UPayloadFormat, UPriority, UResource, UStatus, UUri, UUID,
+};
+
+use protobuf::{Enum, MessageField, SpecialFields};
+
+pub fn convert_json_to_jsonstring<T: serde::Serialize>(value: &T) -> String {
+    if let Ok(json_string) = serde_json::to_string(value) {
+        json_string
+    } else {
+        // Handle the error
+        error!("Error: Failed to convert to JSON string");
+        "None".to_owned()
+    }
+}
+
+pub fn get_ustatus_code(u_status: &UStatus) -> u32 {
```

Comment:
Do we still need to do this?

Isn't this exactly what get_code() does?

Reply:
like most things in rust, it was tricky to figure out, had to use "as i32", but I think I got it working.